src/streaming/controllers/CmcdController.js

```diff
@@ -376,10 +240,103 @@ function CmcdController() {
         }
     }
 
+    function triggerCmcdEventMode(event){
+        const targets = settings.get().streaming.cmcd.targets;
+        const eventModeTargets = targets.filter((target) => target.cmcdMode === Constants.CMCD_MODE.EVENT);
+
+        if (eventModeTargets.length === 0) {
+            return;
+        }
+
+        const cmcdData = cmcdModel.triggerCmcdEventMode(event);
+        
+        eventModeTargets.forEach(targetSettings => {
+            if (targetSettings.enabled) {
+
+                if (targetSettings.events?.length === 0) {
+                    logger.warn('CMCD Event Mode is enabled, but the "events" setting is empty. No event-specific CMCD data will be sent.');
+                }
+
+                let events = targetSettings.events ? targetSettings.events : Object.values(Constants.CMCD_REPORTING_EVENTS);
+
+                if (!events.includes(event)) {
+                    return;
+                }
+
+                let httpRequest = new CmcdReportRequest();
+
+                httpRequest.url = targetSettings.url;
+                httpRequest.type = HTTPRequest.CMCD_EVENT;
+                httpRequest.method = HTTPRequest.GET;
+
+                _updateRequestUrlAndHeadersWithCmcd(httpRequest, cmcdData, targetSettings)
+                sendCmcdDataReport(httpRequest);
+            }
+        });
+    }
+
+    function sendCmcdDataReport(request){
```

Comment:
This is a private method
```suggestion
    function _sendCmcdDataReport(request){
```

Reply:
Fixed